kingpin/actors/aws/iam/entities.py

```diff
@@ -1007,3 +1020,467 @@ def _execute(self):
+                role, entity['arn'], 'role'))
+
+        # Wait until detachment is completed
+        yield tasks
+
+    @gen.coroutine
+    def _ensure_policy_entities(self, policy, desired_entities,
+                                existing_entities, entity_type):
+        """Ensure that the Policy has certain attached entities.
+
+        If we are managing a particular type of entity attachment (user, group,
+        role), then this method will ensure that the appropriate attachments
+        are there, and detach the ones that shouldn't be. If the
+        desired_entities is None, then we do not manage them at all.
+
+        Note: we don't do any entity/attachment state gathering in this method
+        because it would be called many times, and we can get all that data
+        with a single API call to Amazon.
+
+        args:
+            policy: The Entity Policy object -- what comes back from
+            self._get_entity()
+            desired_entities: A list of desired attached entities by name
+            existing_entities: The currently attached entities
+            entity_type: The entity type string (user, role, group)
+        """
+        # Desired entities is None? Then we don't manage them at all -- no
+        # purging, no adding.
+        if desired_entities is None:
+            raise gen.Return()
+
+        # If we were passed in a string (a single entity), turn it into a list
+        if isinstance(desired_entities, basestring):
+            desired_entities = [desired_entities]
+
+        # If we were passed in a Nonetype policy entity object, then that means
+        # the policy doesn't exist. Use a fake name at this point, because we
+        # must be in a Dry run.
+        if policy is None:
+            policy = {'arn': self.option('name')}
+
+        to_attach = set(desired_entities) - set(existing_entities)
+        to_detach = set(existing_entities) - set(desired_entities)
+
+        tasks = []
+        for entity in to_attach:
+            tasks.append(self._attach_entity_policy(
+                entity, policy['arn'], entity_type))
+
+        for entity in to_detach:
+            tasks.append(self._detach_entity_policy(
+                entity, policy['arn'], entity_type))
+
+        yield tasks
+
+    @gen.coroutine
+    def _attach_entity_policy(self, name, policy, entity_type):
+        """Calls the policy attach method.
+
+        There are three attachment methods - one for users, groups and roles.
+
+        args:
+            name: The name of the Instance Profile we're managing
+            policy: The policy ARN we're attaching to
+            entity_type: String name of the entity type (user, group, role)
+        """
+        if self._dry:
+            self.log.warning('Would attach %s %s to %s' %
+                             (entity_type, name, policy))
+            raise gen.Return()
+
+        method = eval('self.iam_conn.attach_%s_policy' % entity_type)
```

Comment:
`eval()` always makes my skin crawl. You don't actually need a dynamic method here. Consider this setup instead:

```python
type_methods = {
  'user': self.iam_conn.attach_user_policy,
  'group': self.iam_conn.attach_group_policy,
  'role': self.iam_conn.attach_role_policy
}

method = type_methods[entity_type]
```